diff --git a/platform/marvell-arm64/platform.conf b/platform/marvell-arm64/platform.conf
index 15f25a2..e82c10b 100644
--- a/platform/marvell-arm64/platform.conf
+++ b/platform/marvell-arm64/platform.conf
@@ -13,8 +13,8 @@ initrd_addr=0x2000000
 VAR_LOG=512
 DISK_SIZE=9
 
-kernel_fname="/boot/vmlinuz-4.9.0-9-2-arm64"
-initrd_fname="/boot/initrd.img-4.9.0-9-2-arm64"
+kernel_fname="/boot/vmlinuz-4.19.0-6-armmp"
+initrd_fname="/boot/initrd.img-4.19.0-6-armmp"
 fdt_fname="/boot/armada-7020-comexpress.dtb"
 fit_fname="/boot/sonic_arm64.fit"
 
diff --git a/files/build_templates/init_cfg.json.j2 b/files/build_templates/init_cfg.json.j2
index 7a0a8d1..4cf4a06 100644
--- a/files/build_templates/init_cfg.json.j2
+++ b/files/build_templates/init_cfg.json.j2
@@ -18,7 +18,7 @@
         }
     },
     "FEATURE": {
-{%- for feature, status in [("sflow", "disabled"), ("telemetry", "enabled")] %}
+{%- for feature, status in [("sflow", "enabled"), ("telemetry", "enabled")] %}
         "{{feature}}": {
             "status": "{{status}}"
         }{% if not loop.last %},{% endif -%}
diff --git a/debian/control b/debian/control
index e2e8857..94926fc 100644
--- a/debian/control
+++ b/debian/control
@@ -9,12 +9,12 @@ Section: net
 
 Package: sonic-mgmt-common
 Priority: extra  
-Architecture: amd64
+Architecture: all
 Depends: ${shlibs:Depends}, ${misc:Depends}  
 Description: SONiC Management Infrastructure
 
 Package: sonic-mgmt-common-codegen
 Priority: extra  
-Architecture: amd64
+Architecture: all
 Depends: ${shlibs:Depends}, ${misc:Depends}  
 Description: SONiC Management Infrastructure generated code
