diff --git a/scripts/generate_dump b/scripts/generate_dump
index 3b669ff..a0793cc 100755
--- a/scripts/generate_dump
+++ b/scripts/generate_dump
@@ -182,7 +182,7 @@ save_nat_info() {
 save_proc() {
     local procfiles="$@"
     $MKDIR $V -p $TARDIR/proc \
-        && $CP $V -r $procfiles $TARDIR/proc \
+        && (for f in $procfiles; do $CP $V -r $f $TARDIR/proc || echo "$f not found" > $TARDIR/$f; done) \
         && $TAR $V -rhf $TARFILE -C $DUMPDIR --mode=+rw $BASE/proc \
         && $RM $V -rf $TARDIR/proc
 }
@@ -511,10 +511,12 @@ main() {
         fi
     done
 
-    # run 'hw-management-generate-dump.sh' script and save the result file
-    /usr/bin/hw-management-generate-dump.sh    
-    save_file "/tmp/hw-mgmt-dump*" "hw-mgmt" false
-    rm -f /tmp/hw-mgmt-dump*
+    if [[ "$asic" = "mellanox" ]]; then
+        # run 'hw-management-generate-dump.sh' script and save the result file
+        /usr/bin/hw-management-generate-dump.sh    
+        save_file "/tmp/hw-mgmt-dump*" "hw-mgmt" false
+        rm -f /tmp/hw-mgmt-dump*
+    fi
 
     # clean up working tar dir before compressing
     $RM $V -rf $TARDIR
